### ISSUE TITLE
Bug - lists: update roles to match React

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly#5385]
**Describe the problem**
Followup to https://github.com/patternfly/patternfly/pull/5357 (and possibly https://github.com/patternfly/patternfly-react/issues/8453)

The following components need roles updated to match React:

- Tabs
- Dropdown
- OptionsMenu
- SearchInput (the list for autocomplete examples)
- OverflowMenu
- ProgressStep may need `role="listitem"` added to it due to the `display:contents` CSS
- ApplicationLauncher and ContextSelector (I had updated them to `role="list"`, but should be menu role; the items will also need `role="menuitem"` added)

Components listed here and previously updated in 5357 could also be updated to have aria-labels added in in the examples.
